kore/src/Kore/Repl/Data.hs

```diff
@@ -388,36 +393,80 @@ initializeProofFor
     :: MonadState (ReplState claim) m
     => Claim claim
     => claim
+    -> ClaimIndex
     -> m ()
-initializeProofFor claim =
+initializeProofFor claim cindex = do
+    gphs <- gets graphs
+    lbls <- gets labels
     modify (\st -> st
-        { graph  = emptyExecutionGraph claim
+        { graphs  = Map.insert cindex (emptyExecutionGraph claim) gphs
         , claim  = claim
+        , claimIndex = cindex
         , node   = ReplNode 0
-        , labels = Map.empty
+        , labels = Map.insert cindex Map.empty lbls
+        })
+
+switchToProof
+    :: MonadState (ReplState claim) m
+    => Claim claim
+    => claim
+    -> ClaimIndex
+    -> m ()
+switchToProof claim cindex = do
+    modify (\st -> st
+        { claim = claim
+        , claimIndex = cindex
+        , node = ReplNode 0
         })
 
 -- | Get the internal representation of the execution graph.
 getInnerGraph
     :: MonadState (ReplState claim) m
+    => Claim claim
     => m InnerGraph
-getInnerGraph = Strategy.graph . graph <$> get
+getInnerGraph =
+    fmap Strategy.graph getExecutionGraph
+
+-- | Get the current execution graph
+getExecutionGraph
+    :: MonadState (ReplState claim) m
+    => Claim claim
+    => m ExecutionGraph
+getExecutionGraph = do
+    cindex <- gets claimIndex
+    gph <- gets graphs
+    cl <- gets claim
```

Comment:
Why not `ReplState { claimIndex, graphs, claim } <- get` instead of these 3 lines?